Rename params: *url to *uri

fsackur opened the pull request:
# PR Summary

- Changed `-URL` params to `-Uri` (including, e.g., `-LicenseUrl` => `-LicenseUri`)
    - Updated (almost) all identifiers in source code `url` => `uri`, for consistency
    - Updated docs
    - A few strings and comments left at `URL`, where they make sense in the context
- Added `test/result*` to `.gitignore`
- Amended `build.ps1` to try importing `PSPackageProject` before installing again

## PR Context

Ref https://github.com/PowerShell/PowerShellGet/issues/460

Broadly speaking, all installed-by-default modules use `uri` in parameter names instead of `url`. To provide consistency, I think PowerShellGet should also follow this convention.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: <!-- Number/link of that issue here -->
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: <!-- Number/link of that issue here -->
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

fsackur commented:
Requesting feedback!

I hope it's not too contentious to change all the terminology in private code; it seemed to me that leaving inconsistencies would be a poor dev experience for future maintainers.

alerickson commented on src/code/PublishPSResource.cs:
```diff
@@ -613,9 +613,9 @@ private bool IsValidModuleManifest(string moduleManifestPath)
+                <licenseUri>https://licenses.nuget.org/MIT</licenseUri>
                 <icon>Powershell_black_64.png</icon>
-                <projectUrl>https://github.com/PowerShell/PowerShell</projectUrl>
+                <projectUri>https://github.com/PowerShell/PowerShell</projectUri>
```
Same comment here-- this should stay as 'projectUrl'

fsackur replied:
Done

alerickson commented on test/PublishPSResource.Tests.ps1:
```diff
@@ -168,62 +168,62 @@ Describe "Test Publish-PSResource" {
         $expectedNuspec = Join-path -Path $expectedExpandedPath -ChildPath "$script:PublishModuleName.nuspec"
         $expectedNuspecContents =  Get-Content -Path $expectedNuspec -Raw
-        $expectedNuspecContents.Contains($projectUrl) | Should Be $true
+        $expectedNuspecContents.Contains($projectUri) | Should Be $true
```
This should stay as 'projectUrl'

fsackur replied:
Done

alerickson commented on test/PublishPSResource.Tests.ps1:
```diff
@@ -168,62 +168,62 @@ Describe "Test Publish-PSResource" {
-        $projectUrl = "https://www.fakeprojecturl.com"
-        Publish-PSResource -Path $script:PublishModuleBase -ProjectUrl $projectUrl
+        $projectUri = "https://www.fakeprojectUri.com"
+        Publish-PSResource -Path $script:PublishModuleBase -ProjectUri $projectUri
```
Lines 214 and 215 should be 'projectUrl'

fsackur replied:
Done

alerickson commented on test/PublishPSResource.Tests.ps1:
```diff
@@ -168,62 +168,62 @@ Describe "Test Publish-PSResource" {
-    
-    It "Publish a module with -ProjectUrl" {
+
+    It "Publish a module with -ProjectUri" {
```
Should be 'ProjectUrl'

fsackur replied:
Done

alerickson commented on test/PublishPSResource.Tests.ps1:
```diff
@@ -168,62 +168,62 @@ Describe "Test Publish-PSResource" {
         $expectedNuspec = Join-path -Path $expectedExpandedPath -ChildPath "$script:PublishModuleName.nuspec"
         $expectedNuspecContents =  Get-Content -Path $expectedNuspec -Raw
-        $expectedNuspecContents.Contains($iconUrl) | Should Be $true
+        $expectedNuspecContents.Contains($iconUri) | Should Be $true
```
Should be 'iconUrl'

fsackur replied:
Done

alerickson commented on test/PublishPSResource.Tests.ps1:
```diff
@@ -168,62 +168,62 @@ Describe "Test Publish-PSResource" {
     }
 
-    It "Publish a module with -IconUrl" {
+    It "Publish a module with -IconUri" {
```
Should be 'iconUrl'

alerickson commented on test/PublishPSResource.Tests.ps1:
```diff
@@ -168,62 +168,62 @@ Describe "Test Publish-PSResource" {
-        $iconUrl = "https://www.fakeiconurl.com"
-        Publish-PSResource -Path $script:PublishModuleBase -IconUrl $iconUrl
+        $iconUri = "https://www.fakeiconuri.com"
+        Publish-PSResource -Path $script:PublishModuleBase -IconUri $iconUri
```
Should be 'iconUrl'

alerickson commented on test/PublishPSResource.Tests.ps1:
```diff
@@ -168,62 +168,62 @@ Describe "Test Publish-PSResource" {
         $expectedNuspec = Join-path -Path $expectedExpandedPath -ChildPath "$script:PublishModuleName.nuspec"
         $expectedNuspecContents =  Get-Content -Path $expectedNuspec -Raw
-        $expectedNuspecContents.Contains($licenseUrl) | Should Be $true
+        $expectedNuspecContents.Contains($licenseUri) | Should Be $true
```
Should be 'licenseUrl'

alerickson commented on test/PublishPSResource.Tests.ps1:
```diff
@@ -168,62 +168,62 @@ Describe "Test Publish-PSResource" {
-        $licenseUrl = "https://www.fakelicenseurl.com"
-        Publish-PSResource -Path $script:PublishModuleBase -LicenseUrl $licenseUrl
+        $licenseUri = "https://www.fakelicenseuri.com"
+        Publish-PSResource -Path $script:PublishModuleBase -LicenseUri $licenseUri
```
Should be 'licenseUrl'

alerickson commented on test/PublishPSResource.Tests.ps1:
```diff
@@ -168,62 +168,62 @@ Describe "Test Publish-PSResource" {
     }
 
-    It "Publish a module with -LicenseUrl" {
+    It "Publish a module with -LicenseUri" {
```
Should be 'licenseUrl'

alerickson commented on src/code/PublishPSResource.cs:
```diff
@@ -587,19 +587,19 @@ private bool IsValidModuleManifest(string moduleManifestPath)
             metadataElementsDictionary.Add("tags", tags);
 
-            if (parsedMetadataHash.ContainsKey("licenseurl"))
+            if (parsedMetadataHash.ContainsKey("licenseuri"))
```
This still needs to remain 'licenseUrl' because we can't change the module manifest metadata.

fsackur replied:
Done

alerickson commented on src/code/PublishPSResource.cs:
```diff
@@ -587,19 +587,19 @@ private bool IsValidModuleManifest(string moduleManifestPath)
+            if (parsedMetadataHash.ContainsKey("licenseuri"))
             {
-                metadataElementsDictionary.Add("licenseUrl", parsedMetadataHash["licenseurl"].ToString().Trim());
+                metadataElementsDictionary.Add("licenseUri", parsedMetadataHash["licenseuri"].ToString().Trim());
```
Again should be 'licenseUrl'

fsackur replied:
Done

alerickson commented on src/code/PublishPSResource.cs:
```diff
@@ -587,19 +587,19 @@ private bool IsValidModuleManifest(string moduleManifestPath)
             }
 
-            if (parsedMetadataHash.ContainsKey("projecturl"))
+            if (parsedMetadataHash.ContainsKey("projecturi"))
```
For the same reason as above, should be 'projectUrl'

fsackur replied:
Done

alerickson commented on src/code/PublishPSResource.cs:
```diff
@@ -587,19 +587,19 @@ private bool IsValidModuleManifest(string moduleManifestPath)
+            if (parsedMetadataHash.ContainsKey("projecturi"))
             {
-                metadataElementsDictionary.Add("projectUrl", parsedMetadataHash["projecturl"].ToString().Trim());
+                metadataElementsDictionary.Add("projectUri", parsedMetadataHash["projecturi"].ToString().Trim());
```
And again, 'projectUrl'

fsackur replied:
Done

alerickson commented on src/code/PublishPSResource.cs:
```diff
@@ -587,19 +587,19 @@ private bool IsValidModuleManifest(string moduleManifestPath)
             }
 
-            if (parsedMetadataHash.ContainsKey("iconurl"))
+            if (parsedMetadataHash.ContainsKey("iconuri"))
```
Again, for the same reason as above should be 'iconUrl'

fsackur replied:
Done

alerickson commented on src/code/PublishPSResource.cs:
```diff
@@ -587,19 +587,19 @@ private bool IsValidModuleManifest(string moduleManifestPath)
+            if (parsedMetadataHash.ContainsKey("iconuri"))
             {
-                metadataElementsDictionary.Add("iconUrl", parsedMetadataHash["iconurl"].ToString().Trim());
+                metadataElementsDictionary.Add("iconUri", parsedMetadataHash["iconuri"].ToString().Trim());
```
Should be 'iconUrl'

fsackur replied:
Done

alerickson commented on src/code/PublishPSResource.cs:
```diff
@@ -613,9 +613,9 @@ private bool IsValidModuleManifest(string moduleManifestPath)
                 <requireLicenseAcceptance>false</requireLicenseAcceptance>
                 <license type="expression">MIT</license>
-                <licenseUrl>https://licenses.nuget.org/MIT</licenseUrl>
+                <licenseUri>https://licenses.nuget.org/MIT</licenseUri>
```
Should be 'licenseUrl'

fsackur replied:
Done

alerickson reviewed (changes requested):
Thank you @fsackur for opening up this PR!  
Just a few things--
1)  There's a handful of changes in PublishPSResource that should actually not be changed, specifically 'licenseUrl', 'iconUrl', and 'projectUrl'.  Those names are specified in the module manifest, which is used to generate the nuspec, so it'll cause problems when publishing (see [nuspec reference](https://docs.microsoft.com/en-us/nuget/reference/nuspec#general-form-and-schema) if interested)

2)  This is very minor but the last two tests (SetPSResourceRepository.Tests.ps1 and UnregisterPSResourceRepository.Tests.ps1) have Uri in all caps whereas the other tests have it in pascal case.  I would recommend changing the former tests to pascal case just to be consistent.  Of course this is just an aesthetic request!

anamnavi commented on test/RegisterPSResourceRepository.Tests.ps1:
```diff
@@ -25,26 +25,26 @@ Describe "Test Register-PSResourceRepository" {
 
-    It "register repository given Name, URL (bare minimum for NameParmaterSet)" {
-        $res = Register-PSResourceRepository -Name "testRepository" -URL $tmpDir1Path -PassThru
+    It "register repository given Name, URI (bare minimum for NameParameterSet)" {
```
we should match cmdlet parameter's casing (i.e Uri). Please also make this change for lines 36, 44, 143, 147, 151, 155, 164, and 224:
```suggestion
    It "register repository given Name, URI (bare minimum for NameParameterSet)" {
```
```suggestion
    It "register repository given Name, Uri (bare minimum for NameParameterSet)" {
```

anamnavi commented on .gitignore:
```diff
@@ -3,3 +3,4 @@ src/code/obj
 srcOld/code/bin
 srcOld/code/obj
 out
+test/result*
```
please remove this change, I imagine it's to ignore local test result files created on your machine.

fsackur replied:
My bad, removed

anamnavi reviewed (changes requested):
Hi @fsackur! Thanks for creating this PR and for diligently going through all the files and making appropriate changes. I know it looks like a lot of comments but I had feedback in 3 main areas:

1) please remove the local `test/result*` file reference from the .gitignore. Thanks :)

2) it's a bit of a nitpick but it would be nice to match the casing of cmdlet's Uri parameter across all tests/code. I know in some of the original code we accidentally had it as "URL" so I understand changing it to "URI" however "Uri" where appropriate would be best. I left suggestions on where all this would happen.

3) I echoed @alerickson's feedback on reverting the *url to *uri change for LicenseUrl, IconUrl, ProjectUrl in a few documentation files just so that we don't miss it.

Thanks for making all these changes, it's looking great :D

fsackur commented:
My thanks for the detail - this represents a lot of review work. I will not be able to look at it for 10 days at least - I aim to update the PR by 2021-12-16.

fsackur commented:
Aplogies for sitting on this so long, after anamnavi put so much time into review. I will update the PR this week

kilasuit commented:
@alerickson / @anamnavi  - It's worth remembering that whilst nuget uses `ProjectURL` & `LicenseURL`, PowerShell Module Manifests have always used `ProjectUri` & `LicenseUri` as their properties as can be seen in the [New-ModuleManifest doc](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/new-modulemanifest?view=powershell-7.2#example-1---create-a-new-module-manifest)

Whilst I haven't looked deeply at this repositories code base (or this PR either) we need to be sure that the module manifest is correct as people who run the following would end up being broken if the resultant manifest uses `Url` not `Uri` in all the properties in the PSData hashtable under the PrivateData HashTable.

Just thought it worth mentioning

fsackur commented on Docs/PublishPSResource.md:
````diff
@@ -8,12 +8,12 @@ It publishes a specified resource from the local computer to an online Nuget-bas
 ### PathParameterSet
 ``` PowerShell
-[[-APIKey] <string>] [-Repository <string>] [-DestinationPath <string>] [-Path] <string>] [-Credential <pscredential>] [-SkipDependenciesCheck] [-ReleaseNotes <string>] [-Tags <string[]>] [-LicenseUrl <string>] [-IconUrl <string>] [-ProjectUrl <string>] [-NuspecPath <string>] [-Proxy <Uri>][-ProxyCredential <pscredential>] [-WhatIf] [-Confirm] [<CommonParameters>]
+[[-APIKey] <string>] [-Repository <string>] [-DestinationPath <string>] [-Path] <string>] [-Credential <pscredential>] [-SkipDependenciesCheck] [-ReleaseNotes <string>] [-Tags <string[]>] [-LicenseUri <string>] [-IconUri <string>] [-ProjectUri <string>] [-NuspecPath <string>] [-Proxy <Uri>][-ProxyCredential <pscredential>] [-WhatIf] [-Confirm] [<CommonParameters>]
````
Docs deleted upstream

fsackur commented on Docs/RegisterPSResourceRepository.md:
```diff
@@ -115,19 +115,19 @@ if `-PassThru` is specified, output is:
 ## Tests
 
-Tests able to register repositories with different allowed URL URI schemes.
+Tests able to register repositories with different allowed URI schemes.
```
Docs deleted in HEAD

fsackur commented on Docs/UnregisterPSResourceRepository.md:
```diff
@@ -33,7 +33,7 @@ none
 ## Tests
 
-Tests able to unregister repositories with different allowed URL URI schemas.
+Tests able to unregister repositories with different allowed URI schemas.
```
Docs deleted in HEAD

fsackur commented:
Thanks for the feedback, I have addressed the points. Can I get a re-review?

@anamnavi you asked me to pascal-case some `URI`s in some strings and comments. I've changed a few. Mainly, I've used `Uri` where the comment refers to the parameter and `URI` when expressing an actual URI. That's the style I've seen the most. It seems more natural in a sentence. Want me to change?

alerickson commented on src/code/RegisterPSResourceRepository.cs:
```diff
@@ -209,9 +209,9 @@ private PSRepositoryInfo AddToRepositoryStoreHelper(string repoName, Uri repoUrl
             }
 
-            if (repoUrl == null || !(repoUrl.Scheme == Uri.UriSchemeHttp || repoUrl.Scheme == Uri.UriSchemeHttps || repoUrl.Scheme == Uri.UriSchemeFtp || repoUrl.Scheme == Uri.UriSchemeFile))
+            if (repoUri == null || !(repoUri.Scheme == System.Uri.UriSchemeHttp || repoUri.Scheme == System.Uri.UriSchemeHttps || repoUri.Scheme == System.Uri.UriSchemeFtp || repoUri.Scheme == System.Uri.UriSchemeFile))
```
Is the `System.`  needed here?

fsackur replied:
Yes, to disambiguate from the `Uri` param at L54

alerickson commented on src/code/Utils.cs:
```diff
@@ -286,7 +286,7 @@ public static string[] GetStringArray(ArrayList list)
             try
             {
-                // This is needed for a relative path urlstring. Does not throw error for an absolute path.
+                // This is needed for a relative path uri string. Does not throw error for an absolute path.
```
```suggestion
                // This is needed for a relative path URI string. Does not throw error for an absolute path.
```

alerickson commented on src/code/SetPSResourceRepository.cs:
```diff
@@ -177,16 +177,16 @@ private PSRepositoryInfo UpdateRepositoryStoreHelper(string repoName, Uri repoUr
+            if (repoName.Equals("PSGallery", StringComparison.OrdinalIgnoreCase) && repoUri != null)
             {
-                throw new ArgumentException("The PSGallery repository has a pre-defined URL.  Setting the -URL parameter for this repository is not allowed, instead try running 'Register-PSResourceRepository -PSGallery'.");
+                throw new ArgumentException("The PSGallery repository has a pre-defined URI. Setting the -Uri parameter for this repository is not allowed, instead try running 'Register-PSResourceRepository -PSGallery'.");
```
```suggestion
                throw new ArgumentException("The PSGallery repository has a pre-defined Uri. Setting the -Uri parameter for this repository is not allowed, instead try running 'Register-PSResourceRepository -PSGallery'.");
```

alerickson commented on src/code/SetPSResourceRepository.cs:
```diff
@@ -177,16 +177,16 @@ private PSRepositoryInfo UpdateRepositoryStoreHelper(string repoName, Uri repoUr
 
-            // check PSGallery URL is not trying to be set
-            if (repoName.Equals("PSGallery", StringComparison.OrdinalIgnoreCase) && repoUrl != null)
+            // check PSGallery URI is not trying to be set
```
```suggestion
            // check PSGallery Uri is not trying to be set
```

alerickson commented on src/code/SetPSResourceRepository.cs:
```diff
@@ -162,11 +162,11 @@ protected override void ProcessRecord()
+            if (repoUri != null && !(repoUri.Scheme == System.Uri.UriSchemeHttp || repoUri.Scheme == System.Uri.UriSchemeHttps || repoUri.Scheme == System.Uri.UriSchemeFtp || repoUri.Scheme == System.Uri.UriSchemeFile))
             {
-                throw new ArgumentException("Invalid url, must be one of the following Uri schemes: HTTPS, HTTP, FTP, File Based");
+                throw new ArgumentException("Invalid URI, must be one of the following URI schemes: HTTPS, HTTP, FTP, File Based");
```
```suggestion
                throw new ArgumentException("Invalid Uri, must be one of the following Uri schemes: HTTPS, HTTP, FTP, File Based");
```

alerickson commented on src/code/RepositorySettings.cs:
```diff
@@ -119,14 +119,14 @@ public static PSRepositoryInfo Add(string repoName, Uri repoURL, int repoPriorit
 
         /// <summary>
-        /// Updates a repository name, URL, priority, installation policy, or credential information
+        /// Updates a repository name, URI, priority, installation policy, or credential information
```
```suggestion
        /// Updates a repository name, Uri, priority, installation policy, or credential information
```

alerickson commented on help/Register-PSResourceRepository.md:
```diff
@@ -149,9 +149,9 @@ Accept pipeline input: False
+### -Uri
 Specifies the location of the repository to be registered.
-URL can be of the following Uri schemas: HTTPS, HTTP, FTP, file share based.
+URI can be of the following URI schemas: HTTPS, HTTP, FTP, file share based.
```
```suggestion
Uri can be of the following Uri schemas: HTTPS, HTTP, FTP, file share based.
```

alerickson commented on help/Register-PSResourceRepository.md:
```diff
@@ -225,8 +225,8 @@ This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable
+Registering the PSGallery repository must be done via the PSGalleryParameterSet (i.e by using the 'PSGallery' parameter instead of 'Name' and 'Uri' parameters).
 
-URL string input must be of one of the following Uri schemes: HTTP, HTTPS, FTP, File
+URI string input must be of one of the following URI schemes: HTTP, HTTPS, FTP, File
```
```suggestion
Uri string input must be of one of the following Uri schemes: HTTP, HTTPS, FTP, File
```

alerickson reviewed (changes requested):
Looks good, just a couple of spots that both @anamnavi and I added suggested changes to be consistent with the Pascal casing of 'Uri'.  Once those are in this should be good to go 😄

fsackur commented on src/code/RegisterPSResourceRepository.cs:
```diff
@@ -209,9 +209,9 @@ private PSRepositoryInfo AddToRepositoryStoreHelper(string repoName, Uri repoUrl
+            if (repoUri == null || !(repoUri.Scheme == System.Uri.UriSchemeHttp || repoUri.Scheme == System.Uri.UriSchemeHttps || repoUri.Scheme == System.Uri.UriSchemeFtp || repoUri.Scheme == System.Uri.UriSchemeFile))
             {
-                throw new ArgumentException("Invalid url, must be one of the following Uri schemes: HTTPS, HTTP, FTP, File Based");
+                throw new ArgumentException("Invalid URI, must be one of the following URI schemes: HTTPS, HTTP, FTP, File Based");
```
```suggestion
                throw new ArgumentException("Invalid Uri, must be one of the following Uri schemes: HTTPS, HTTP, FTP, File Based");
```

fsackur commented on src/code/RegisterPSResourceRepository.cs:
```diff
@@ -335,19 +335,19 @@ private PSRepositoryInfo RepoValidationHelper(Hashtable repo)
                 WriteError(new ErrorRecord(
-                        new PSInvalidOperationException("Repository url cannot be null"),
-                        "NullURLForRepositoriesParameterSetRegistration",
+                        new PSInvalidOperationException("Repository URI cannot be null"),
```
```suggestion
                        new PSInvalidOperationException("Repository Uri cannot be null"),
```

fsackur commented:
I take that feedback to mean that `URI` should always be `Uri` in comments and strings, so I have changed all occurrences in https://github.com/PowerShell/PowerShellGet/pull/551/commits/6c92c80e16fcdec4b767b32a98b24f4cafe715be

alerickson commented:
Thanks for giving us a heads up on that @kilasuit!  I'm going to open up an issue so that we can test and make sure we have the 'projectUrl' or 'projectUri' and 'iconUrl' or 'iconUri' in the appropriate places.  I'll open up a new PR to fix any changes there and add you to review it (no pressure to if you'd rather not).

fsackur commented:
I did publish a module to the gallery from this branch. I was then able to find it; it did have project and licese properties as set in my .psd1.

fsackur commented:
@anamnavi thanks for all the work you and @alerickson have put into this review. I have taken your review comments to mean that I should change `URI` => `Uri` in ALL comments and strings, so I've done that in ALL occurrences (there were a few that you hadn't highlighted). 

The only places where all-caps `URI` still appear:

```
Path                       LineNumber Line
----                       ---------- ----
.\CHANGELOG.md                     38 - Fix bug to register repositories with local file share paths, ensuring repositories with valid URIs can be registered.
.\test\PSGetModuleInfo.xml         34       <URI N="LicenseUri">https://github.com/PowerShell/SecretManagement/blob/master/LICENSE</URI>
.\test\PSGetModuleInfo.xml         35       <URI N="ProjectUri">https://github.com/powershell/secretmanagement</URI>
.\test\PSGetTestUtils.psm1        514 .LICENSEURI$(if ($LicenseUri) {" $LicenseUri"})
.\test\PSGetTestUtils.psm1        516 .PROJECTURI$(if ($ProjectUri) {" $ProjectUri"})
.\test\PSGetTestUtils.psm1        518 .ICONURI$(if ($IconUri) {" $IconUri"})
```

I can haz merge?

anamnavi commented on src/PowerShellGet.psd1:
```diff
@@ -46,5 +46,5 @@ See change log (CHANGELOG.md) at https://github.com/PowerShell/PowerShellGet
     }
 
-    HelpInfoURI       = 'http://go.microsoft.com/fwlink/?linkid=855963'
+    HelpInfoUri       = 'http://go.microsoft.com/fwlink/?linkid=855963'
```
@fsackur can you please revert this change specifically? Like just change it back to "HelpInfoURI".
This variable is uppercased "URI" (i.e HelpInfoURI) in the module manifest files for packages on PSGallery. Example: https://www.powershellgallery.com/packages/PowerShellGet/2.2.5/Content/PowerShellGet.psd1   L286).

alerickson replied:
Actually I just noticed 'ProjectUri' and 'LicenseUri' both have Uri in Pascal case, so I think we should leave this as is for now

anamnavi replied:
okay that makes sense. I'll merge the PR in then :)

anamnavi commented:
@fsackur thank you for the hard work and being so diligent with these changes and also sharing the rule of thumb you used when changing URI => Uri in all strings and comments, that's helpful for us to know! Sorry for the late response and review, I was a little out of it after my booster shot. This looks good to me, only one minor change request, per my comment right above this one (it's just a capitalization change for consistency sake with other .psd1 files, and is just a small nitpick if you don't mind addressing that). Thank you for your contribution! :D
